### DISCUSSION TITLE
Search can't search C code in cpython even though files are indexed

### DISCUSSION THREAD

**Discussion by DefaultRyan** [community/community#37569] [Other Feature Feedback, Questions, & Ideas] [answered]
I keep having issues searching code in the https://github.com/python/cpython repository. For instance, searching for `win32_xtat_impl` using this query (https://github.com/search?q=win32_xstat_impl+repo%3Apython%2Fcpython&type=Code) is not returning any results in the C language.

However, that symbol does exist in the repo at Modules/posixmodule.c, https://github.com/python/cpython/blob/3e07f827b359617664ad0880f218f17ae4483299/Modules/posixmodule.c#L1828

And doing a C language search on the repo shows that posixmodule.c was indexed recently:
https://github.com/search?q=repo%3Apython%2Fcpython+path%3AModules+language%3AC&type=Code

Is something in GitHub search broken, or is there a repository setting that could somehow defeat code search?

**Comment by ossanna16** [comment]
Hi there @DefaultRyan and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by look** [comment, marked as answer]
@DefaultRyan that file is over 350 KiB, which is our limit for indexing at this time. Since it's excluded from the the index, it doesn't appear in results.

The limit is designed to protect our systems, but as this is definitely a useful file, I'll mention it to the team.

**Reply by DefaultRyan** [reply]
Thanks @look. Is there any way to know if there are updates or changes? (Is this specific file going to get an exception, or is there some other repo-based filter that can be applied)? What about other files in that repo?

**Reply by look** [reply]
This is a global limit for GitHub code search (both the old and new systems). If we change the limit, we'll post in this forum and/or in the [changelog](https://github.blog/changelog/).
